docs: Updated GettingStarted docs for Renovate CE and EE (#25284)
renovatebot/renovate: docs/usage/getting-started/running.md

As end user, you can choose from these ways to run Renovate:

- You use the Mend Renovate App

> - You self-host Renovate - e.g. Renovate CLI, Renovate Docker image, or Renovate Community Edition / Enterprise Edition

Reviewer: I think the list of methods to self-host Renovate should be moved somewhere else in the document.

The list item is hard to read because there's too much information in a single line.

Author: I might come back and refactor this page. For now, I would love to have the out-of-date OnPrem info updated to reflect CE/EE.

> - You self-host Renovate - e.g. Renovate CLI, Renovate Docker image, or Renovate Community Edition / Enterprise Edition

Reviewer 2: ```suggestion
- You self-administer/host your own Renovate instance
```

> - You self-host Renovate - eg. Renovate CLI, Renovate Docker image, or Renovate Community Edition / Enterprise Edition containers

Reviewer 2: ```suggestion
- You self-host Renovate - e.g. Renovate CLI, Renovate Docker image, or Renovate Community Edition / Enterprise Edition
```

Need to use image/container terminology consistently

If you're self-hosting Renovate on Windows, read [Self-hosting on Windows](./installing-onboarding.md#self-hosting-on-windows) to prevent line endings from confusing Renovate bot.

> If you're running Renovate Community Edition or Renovate Enterprise Edition, refer to the documentation on the [`mend/renovate-ce-ee` GitHub repository ](https://github.com/mend/renovate-ce-ee).

Reviewer: ```suggestion
If you're running Renovate Community Edition or Renovate Enterprise Edition, refer to the documentation on the [`mend/renovate-ce-ee` GitHub repository](https://github.com/mend/renovate-ce-ee).
```

This should fix the Markdown lint error.

> If you're running Renovate Community Edition or Renovate Enterprise Edition, refer to the documentation on the public GitHub repository [`mend/renovate-ce-ee`](https://github.com/mend/renovate-ce-ee).

Reviewer: ```suggestion
If you're running Renovate Community Edition or Renovate Enterprise Edition, refer to the documentation on the [`mend/renovate-ce-ee` GitHub repository ](https://github.com/mend/renovate-ce-ee).
```

We can drop the word `public`. The reader will only expect public repositories anyway. 😉

Plus I'm rewriting the link fragment so it's clear where you'll end up.

This supports both `gitlab.com` and self-hosted GitLab.
Details for how it works can be found in the project.

#### Mend Renovate Community Edition / Enterprise Edition

> Mend Renovate Community Edition (Renovate CE) and Enterprise Edition (Renovate EE) are commercial offerings of Renovate for self-hosted users.

Reviewer 2: ```suggestion
Mend Renovate Community Edition (Renovate CE) and Enterprise Edition (Renovate EE) are non-Open Source offerings of Renovate for self-hosted users.
```

I'm not sure the best way to phrase this, but the term "commercial" usually implies that it costs money, while CE does not

> Mend Renovate Community Edition (Renovate CE) and Enterprise Edition (Renovate EE) are non-Open Source offerings of Renovate for self-hosted users.

Reviewer: ```suggestion
Mend Renovate Community Edition (Renovate CE) and Enterprise Edition (Renovate EE) are closed-source offerings of Renovate for self-hosted users.
```

I don't like `non-Open Source`. How about  `closed-source` or `proprietary`?

Reviewer 2: Alternatively "EULA-licensed"

Author: I like `proprietary`

> Renovate CE and Renovate EE have support for GitHub (both `github.com` and GitHub Enterprise Server) as well as GitLab self-hosted.

Reviewer: ```suggestion
You can run Renovate CE and Renovate EE on GitHub (both `github.com` and GitHub Enterprise Server) or on a self-hosted GitLab instance.
```

How about something like this?

Author: Thanks for the suggestion. I worded it this way because Renovate doesn't technically "run on" GitHub or GitLab. It just "accesses" repos on GitHub or GitLab. So it "has support for" GH and GL. I might keep this as it is for now.

- It is released every 1-2 months in a slower, more stable cadence than Renovate OSS, which releases on every commit
- It's licensed using an end-user license agreement (EULA) and not the Affero General Public License (AGPL)

Plus, the Enterprise Edition has:

> - Unlimited horizontal scaling to run multiple 'worker' containers

Reviewer 2: ```suggestion
- Horizontal scaling to run multiple 'worker' containers
```

- Dedicated support from Mend.io

> - Premium features, including Smart Merge Control

Reviewer: Is there a link to the Smart Merge Control documentation? From reading just the docs it's not clear what this is, or why I might want to have the feature. 😄

Author: Will consider adding this later.

> Note: Visit the Mend.io website to learn more about [Renovate Enterprise Edition](https://www.mend.io/renovate-enterprise/).

Reviewer: ```suggestion
Go to the Mend.io website to learn more about [Renovate Enterprise Edition](https://www.mend.io/renovate-enterprise/).
```

> For help configuring Renovate CE or Renovate EE, refer to the documentation on the public GitHub repository [`mend/renovate-ce-ee`](https://github.com/mend/renovate-ce-ee).

Reviewer: ```suggestion
To learn how to configure Renovate CE or Renovate EE, read the documentation on the public GitHub repository [`mend/renovate-ce-ee`](https://github.com/mend/renovate-ce-ee).
```

`For help ...` implies that you'll get help from us. But what we really mean is that the reader should go read the docs. So we should say that instead.
